server/app/utils.py

```diff
@@ -563,16 +564,17 @@ def make_zip_filename(user, now):
     return filename+'.zip'
 
 
-def subms_to_gcs(SearchAPI, SubmissionAPI, Submission, user, data, datetime):
-    """
-    Writes all submissions for a given search query
-    to a GCS zip file.
-    """
-    results = SearchAPI.results(data)
+def subms_to_gcs(SearchAPI, SubmissionAPI, Submission, user, data, datetime,
+                start_cursor=None):
+    """Writes all submissions for a given search query to a GCS zip file."""
     zipfile_str, zipfile = start_zip()
-    subm = SubmissionAPI()
-    for result in results:
-        zipfile = add_subm_to_zip(subm, Submission, zipfile, result)
+    subm, has_more, next_cursor = SubmissionAPI(), True, None
```

Comment:
Super Nitpicky (feel free to ignore) - the ordering of the variables should probably be consistent between 571 & 574